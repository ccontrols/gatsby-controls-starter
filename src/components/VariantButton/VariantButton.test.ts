import * as path from "path";
import { loadConfigurations } from "@component-controls/config";
import { renderExample } from "@component-controls/test-renderers";
import { render, act } from "@testing-library/react";
import { renderErr } from "@component-controls/test-renderers";

import doc, {
  overview,
  primary,
  accent,
  disabled,
  success,
  error,
  warning
} from "./VariantButton.docs";

describe("VariantButton", () => {
  const configPath = path.resolve(__dirname, "../../../.config");
  const config = loadConfigurations(configPath);

  test("overview", () => {
    const example = overview;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("primary", () => {
    const example = primary;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("accent", () => {
    const example = accent;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("disabled", () => {
    const example = disabled;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("success", () => {
    const example = success;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("error", () => {
    const example = error;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });

  test("warning", () => {
    const example = warning;

    let rendered;
    act(() => {
      rendered = renderExample({
        example,
        doc,
        config
      });
    });
    if (!rendered) {
      renderErr();
      return;
    }
    const { asFragment } = render(rendered);
    expect(asFragment()).toMatchSnapshot();
  });
});
