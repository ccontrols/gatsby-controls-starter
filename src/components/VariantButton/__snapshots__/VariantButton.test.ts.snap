// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantButton accent snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(155, 81, 224); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Accent
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton disabled snapshot 1`] = `
<DocumentFragment>
  <button
    disabled=""
    style="color: rgb(130, 130, 130); background-color: rgb(224, 224, 224); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Disabled
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton error snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(235, 87, 87); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Error
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 0 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Leone Kirlin
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 1 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Winston Schumm
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 2 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Concepcion Mayer
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 3 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Darion Adams
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 4 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Adalberto Bartell
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton overview 5 snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      style="display: flex; align-items: center;"
    >
      <svg
        aria-hidden="true"
        class="octicon"
        fill="currentColor"
        height="16"
        role="img"
        style="display: inline-block; user-select: none; vertical-align: text-bottom;"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M3.47 7.78a.75.75 0 010-1.06l4.25-4.25a.75.75 0 011.06 0l4.25 4.25a.75.75 0 01-1.06 1.06L9 4.81v7.44a.75.75 0 01-1.5 0V4.81L4.53 7.78a.75.75 0 01-1.06 0z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Jayce Schaden
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton primary snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(47, 128, 237); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Primary
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton success snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(33, 150, 83); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Success
    </div>
  </button>
</DocumentFragment>
`;

exports[`VariantButton warning snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: rgb(242, 242, 242); background-color: rgb(235, 87, 87); border-radius: 8px; border: 1px solid #000000; display: flex; align-items: center;"
  >
    <div
      data-testid="label"
      style="padding: 5px 10px;"
    >
      Warning
    </div>
  </button>
</DocumentFragment>
`;
